### DISCUSSION TITLE
attn kernel 读取 kv cache 时，prefill 用了 LinearIter，decode 用了 BlockIter，这种设计是出于什么考虑呢?

### DISCUSSION THREAD

**Discussion by Time-Limit** [InternLM/lmdeploy#2518] [General]
attn kernel 读取 kv cache 时，prefill 用了 LinearIter，decode 用了 BlockIter，这种设计是出于什么考虑呢?

**Comment by lzhangzz** [comment]
Attention 用 BlockIter 在短 context 上比 LinearIter 上稍微好一点点，但长 context 要差不少
